libs/state-machine/README.md

```diff
@@ -1,6 +1,19 @@
 # state-machine
 
-This library was generated with [Nx](https://nx.dev).
+The State-machine is library to support "Wizard" user interfaces where the user is guided through a series of step which impsct the state of a server-side application, such as entering or modifying data or instructing the application to take an action.
+
+It exports:
+
+-   a set of types that can be imported into other libraries to define `WizardLayout`s for specific workflows.
+-   a handler function to be called on the server side to validiate the user input against the parameters defined in the `WizardLayout`, dispatch and resulting actions and return the state information for the next step to the client
+
+The aim is to keep this library generic, so it could be used for `WizardLayout` describing any workflow and any data model.
+
+## Diagram
+
+[![](https://mermaid.ink/img/pako:eNqdlD1v3DAMhv8KoaFLLujuAlmapUOmQzYtrEWficqSQ9FJDkH-e2UrV-sa5wPxYlt8H5J4TfPJtNGRaUyiu4lCS9eMB8HBhhFFueURg8JtIrHBBsjXT8-Ujy6vri72JPckDcjMJgUOrIwekqJSURfJor5Gxb1GoRVw-ajo_gX_S6yThEq2pittNOc1AYODjiWnTkrjq4ZP0DS6WX37Cx5Y-zOiMD7GEQ4RtJc4HfollkpovmY_6ib25KlVwFY5BvieE3oPXZRhRTZsG6LjjltcoBGPPqJbgcq5E3CPnpfGa3IlfgvhH-Bu_R4JOmSfTmBW_1jldZHanBfTi6HfgESiwEAp4YHO6Q1nHafR43Fb_wXb3rAu0MMH9lHY9nJjCrfdfHciC0KuGs3t8TwTLuMZ6LGezs9N6AYUYo7HuVxhdrD2OBJqvmHb58raE1jT5R8l5VdrqjyLSWZnBpIB2eU18DQHrMnMQNY0-dFRh5NXa2x4zlKcNO6PoTWNykQ7Uxp92Rqm6dCnfEqOs3U3ZbUsG-b5L4HncGU?type=png)](https://mermaid.live/edit#pako:eNqdlD1v3DAMhv8KoaFLLujuAlmapUOmQzYtrEWficqSQ9FJDkH-e2UrV-sa5wPxYlt8H5J4TfPJtNGRaUyiu4lCS9eMB8HBhhFFueURg8JtIrHBBsjXT8-Ujy6vri72JPckDcjMJgUOrIwekqJSURfJor5Gxb1GoRVw-ajo_gX_S6yThEq2pittNOc1AYODjiWnTkrjq4ZP0DS6WX37Cx5Y-zOiMD7GEQ4RtJc4HfollkpovmY_6ib25KlVwFY5BvieE3oPXZRhRTZsG6LjjltcoBGPPqJbgcq5E3CPnpfGa3IlfgvhH-Bu_R4JOmSfTmBW_1jldZHanBfTi6HfgESiwEAp4YHO6Q1nHafR43Fb_wXb3rAu0MMH9lHY9nJjCrfdfHciC0KuGs3t8TwTLuMZ6LGezs9N6AYUYo7HuVxhdrD2OBJqvmHb58raE1jT5R8l5VdrqjyLSWZnBpIB2eU18DQHrMnMQNY0-dFRh5NXa2x4zlKcNO6PoTWNykQ7Uxp92Rqm6dCnfEqOs3U3ZbUsG-b5L4HncGU)
```

Comment:
minor typo on the diagram: "**is** fails validation"

Reply:
updating! glad I save the link to the editor :)